translation-web/src/main/java/eu/europeana/api/translation/web/service/RedisCacheService.java

```diff
@@ -0,0 +1,58 @@
+package eu.europeana.api.translation.web.service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RedisCacheService {
+
+  private RedisTemplate<Integer, String> redisTemplate;
+
+  @Autowired
+  public RedisCacheService(RedisTemplate<Integer, String> redisTemplate) {
+    this.redisTemplate = redisTemplate;
+  }
+
+  /**
+   * Returns a list of Objects (strings) that exist in the cache. If no cache is found for the given key,
+   * the corresponding element in the return list will be null.
+   * @param sourceLang
+   * @param targetLang
+   * @param texts
+   * @return
+   */
+  public List<String> getRedisCache(String sourceLang, String targetLang, List<String> texts) {
+    Collection<Integer> keys = new ArrayList<>();
+    texts.stream().forEach(text -> {
+      String key = text + sourceLang + targetLang;
```

Comment:
we need a dedicated utility method to build the keys, in the spike ticket we need to conclude of the algorithm to be used for generating the hash and the method to convert efficiently the numeric value  to a compact string representtaion